### ISSUE TITLE
Inserting serialized POJO in script tag

### ISSUE THREAD

**Issue by dagnelies** [thymeleaf/thymeleaf#348]
I would like to insert a POJO (plain old java object) as normal Json within a `<script>` tag. ...but how?!?

Given an object like this:

```
class Data {
   String a;
   int b;
   Object c;
}
```

I'd like to obtain something like this as rendered template:

```
< script>
   var data = {
      a: "foo",
      b: 123,
      c: {...}
    }
</script>
```

Until, I haven't found a way to render this using thymeleaf.

I've tried:

```
< script th:inline="javascript">
    var data = /*[[${myDataObj}]]*/ {};
</script>
```

But this failed because apparently thymeleaf is apparently incapable of serializing POJOs.

Then, I've also tried to provide the object already serialized as json:

```
< script th:inline="javascript">
    var data = /*[[${myDataAsJson}]]*/ {};
</script>
```

but this doesn't work either because it'll escape the string again. Again, there seems no way in thymeleaf to provide row unescaped strings.

...now this sucks 

Apparently, this is an issue since 2012 but it seems to rot in a corner:
- https://github.com/thymeleaf/thymeleaf/issues/12
- https://github.com/thymeleaf/thymeleaf/issues/81

...my question is: in the meantime, is there any workaround for this? To serialize a POJO using thymeleaf? It kind of sucks badly.

Thanks

**Comment by beamhead** [comment]
Workaround I used:

    <script type="text/javascript" th:inline="text">
    /*<![CDATA[*/
	    var data= [[${myDataAsJson}]];
    /*]]>*/
    </script>

Works on TL 2.1.2 (not on 2.1.3)

**Comment by ph4nt0ma5ter** [comment]
+1

**Comment by amr** [comment]
This works for me:

In my view:

``` html
<script th:inline="javascript">
/*<![CDATA[*/
  var config = /*[[${config}]]*/ {name: "Sebastian"};
/*]]>*/
</script>

<script type="text/javascript">
  alert('Hey, ' + config.name);
</script>
```

In controller (Spring):

``` java
  public String home(ModelMap model) {
    Config config = new Config();
    config.setName("Amr");

    model.addAttribute("config", config);
    return "index";
  }

  public static class Config {
    private String name;

    public String getName() {
      return name;
    }

    public void setName(String name) {
      this.name = name;
    }
  }
```

Rendered in browser, the relevant part renders as:

``` html
<script>
/*<![CDATA[*/
  var config = {'name':'Amr'};
/*]]>*/
</script>
```

and my alert works fine, displaying "Hey, Amr"

**Comment by dagnelies** [comment]
Hi @amr, 

This thread is about serializing POJOs.
Your example works because it's a "Bean" (with getters/setters), not a POJO (with just members).

**Comment by yarix** [comment]
@dagnelies , your observation between Beans and POJOs saved me hours!! thanks!

**Comment by f0def** [comment]
+1

**Comment by danielfernandez** [comment]
This should be covered by Jackson support in #435, which has just been implemented into `3.0.0-SNAPSHOT`
